### DISCUSSION TITLE
Is it possible to enable/disable trace_log without server restart

### DISCUSSION THREAD

**Discussion by sxm51** [ClickHouse/ClickHouse#70666] [Q&A]
Hi,

Is it possible to enable/disable trace_log without server restart

Found advice in issues pointing towards https://kb.altinity.com/altinity-kb-setup-and-maintenance/altinity-kb-server-config-files/#settings--restart.
While helpful, it doesn't say about trace_log, asynchronous_insert_log etc. Is all part of _logging_.

**Comment by SaltTan** [comment]
If you want to prevent ClickHouse from creating the 'log' tables you need to remove their definition from the config file and this requires a restart https://kb.altinity.com/altinity-kb-setup-and-maintenance/altinity-kb-system-tables-eat-my-disk/

However for 'trace_log' there are settings that you can configure through the default profile and they are applied on the fly https://clickhouse.com/docs/en/operations/system-tables/trace_log

> ClickHouse creates this table when the [trace_log](https://clickhouse.com/docs/en/operations/server-configuration-parameters/settings#trace_log) server configuration section is set. Also see settings: [query_profiler_real_time_period_ns](https://clickhouse.com/docs/en/operations/settings/settings#query_profiler_real_time_period_ns), [query_profiler_cpu_time_period_ns](https://clickhouse.com/docs/en/operations/settings/settings#query_profiler_cpu_time_period_ns), [memory_profiler_step](https://clickhouse.com/docs/en/operations/settings/settings#memory_profiler_step), [memory_profiler_sample_probability](https://clickhouse.com/docs/en/operations/settings/settings#memory_profiler_sample_probability), [trace_profile_events](https://clickhouse.com/docs/en/operations/settings/settings#trace_profile_events).
